emcc.py

```diff
@@ -2369,11 +2388,17 @@ def generate_html(target, options, js_target, target_basename,
   } else {
     // note: no support for code mods (PRECISE_F32==2)
     console.log('running code on the main thread');
+    var filename = '%s';
+    var fileBytes = tryParseAsDataURI(filename);
```

Comment:
That's similar to something @juj pointed out [here](https://github.com/kripken/emscripten/pull/5296#discussion_r127447848); at the moment we could restrict it to `SINGLE_FILE || MEM_INIT_METHOD == 0`, which might make more sense to store in its a dedicated setting if it's expected to potentially change in the future.

We could also save some space by only including `intArrayFromBase64` and `tryParseAsDataURI` in arrayUtils.js under the same conditions.